tests/base/src/test/java/org/wildfly/security/dynamic/ssl/DynamicSSLContextTest.java

```diff
@@ -0,0 +1,354 @@
+                            checkOutputIsOK(clientSocketWithDynamicDefaultSSLContext);
+                            clientSocketWithDynamicDefaultSSLContext.close();
+                        } catch (Exception e) {
+                            Assert.fail();
+                        }
+                    }
+            );
+
+        } catch (Exception e) {
+            e.printStackTrace();
+            Assert.fail();
+        }
+    }
+
+    @Test
+    public void testCreateSocketbyHostAndPortAndConfiguredSSLParams2() {
+        AuthenticationContext context = getAuthenticationContext("wildfly-config-dynamic-ssl-test.xml");
+        context.run(() -> {
+            try {
+                DynamicSSLContext dynamicSSLContext = new DynamicSSLContext();
+                dynamicSSLContext.getDefaultSSLParameters().setCipherSuites(new String[]{"TLS_RSA_WITH_AES_128_CBC_SHA256"});
+                DynamicSslSocketFactory dynamicSSLContextSocketFactory = (DynamicSslSocketFactory) dynamicSSLContext.getSocketFactory();
+                SSLSocket clientSslSocket = (SSLSocket) dynamicSSLContextSocketFactory.createSocket();
+                SSLParameters sslParameters = clientSslSocket.getSSLParameters();
+                sslParameters.setCipherSuites(new String[]{"TLS_RSA_WITH_AES_128_CBC_SHA256"});
+                clientSslSocket.setSSLParameters(sslParameters);
+                dynamicSSLContext.getDefaultSSLParameters().setCipherSuites(new String[]{"TLS_RSA_WITH_AES_128_CBC_SHA256"});
+                clientSslSocket.connect(new InetSocketAddress("localhost", 10000));
+                clientSslSocket.startHandshake();
+                Assert.assertEquals("TLS_RSA_WITH_AES_128_CBC_SHA256", clientSslSocket.getSession().getCipherSuite());
+                checkOutputIsOK(clientSslSocket);
+                clientSslSocket.close();
+            } catch (Exception e) {
+                e.printStackTrace();
+                Assert.fail();
+            }
+        });
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void checkExceptionThrownClientSessionContext() throws Exception {
+        SSLContext sslContext = new DynamicSSLContext();
+        sslContext.getClientSessionContext();
+    }
+
+    @Test(expected = UnsupportedOperationException.class)
+    public void checkExceptionThrownServerSessionContext() throws Exception {
+        SSLContext sslContext = new DynamicSSLContext();
+        sslContext.getServerSessionContext();
+    }
+
+    // thorough testing of sslEngine would need a lot of code with socket implementation that is pretty low level
+    // it is reasonable to assume that it is being tested anyway since sockets created by SSLSocketFactory seem to always use this SSLEngine
+    // here I at least test that the SSLEngine was created with correct host and port
+    @Test
+    public void smokeTestCorrectSSLEngineIsUsed() throws NoSuchAlgorithmException {
```

Comment:
I also wanted to write more thorough tests that use SSLEngine specifically but it required me to add quite big low level classes with socket implementation. I think that it is reasonable to assume that this SSLEngine is tested since sockets created by SSLSocketFactory seem to always use this SSLEngine, so I did not added them.